be/src/exprs/binary_predicate.cpp

```diff
@@ -151,49 +153,108 @@ class ArrayPredicate final : public Predicate {
     EvalCmpZero _comparator;
 };
 
-// add this class because VectorizedNullSafeEqPredicate can't deal with complex data type in columnViewer
-class VectorizedNullSafeArrayEqPredicate final : public Predicate {
+class CommonEqualsPredicate final : public Predicate {
 public:
-    explicit VectorizedNullSafeArrayEqPredicate(const TExprNode& node) : Predicate(node) {}
-    ~VectorizedNullSafeArrayEqPredicate() override = default;
+    explicit CommonEqualsPredicate(const TExprNode& node) : Predicate(node) {}
+    ~CommonEqualsPredicate() override = default;
 
-    Expr* clone(ObjectPool* pool) const override { return pool->add(new VectorizedNullSafeArrayEqPredicate(*this)); }
+    Expr* clone(ObjectPool* pool) const override { return pool->add(new CommonEqualsPredicate(*this)); }
+
+    StatusOr<ColumnPtr> evaluate_checked(ExprContext* context, Chunk* chunk) override {
+        ASSIGN_OR_RETURN(auto l, _children[0]->evaluate_checked(context, chunk));
+        ASSIGN_OR_RETURN(auto r, _children[1]->evaluate_checked(context, chunk));
+
+        if (l->only_null() || r->only_null()) {
+            return ColumnHelper::create_const_null_column(l->size());
+        }
+        auto& const1 = FunctionHelper::get_data_column_of_nullable(l);
+        auto& const2 = FunctionHelper::get_data_column_of_nullable(r);
+
+        size_t lstep = const1->is_constant() ? 0 : 1;
+        size_t rstep = const2->is_constant() ? 0 : 1;
```

Comment:
Seems unnecessary, const column can handle well on accessing i-th element

Reply:
no, like `nullable_column.equals(const column)`,  the implement of most column can't handle it